odoo/tools/config.py

```diff
@@ -137,9 +137,23 @@ def __init__(self, fname=None):
                          help="Listen port for the longpolling HTTP service", type="int", metavar="PORT")
         group.add_option("--no-http", dest="http_enable", action="store_false", my_default=True,
                          help="Disable the HTTP and Longpolling services entirely")
+
+        # HTTP: configure werkzeug proxy mode
+        # https://werkzeug.palletsprojects.com/en/0.16.x/middleware/proxy_fix/
         group.add_option("--proxy-mode", dest="proxy_mode", action="store_true", my_default=False,
                          help="Activate reverse proxy WSGI wrappers (headers rewriting) "
                               "Only enable this when running behind a trusted web proxy!")
+        group.add_option("--proxy-x-for", dest="proxy_x_for", type="int", my_default=1,
+                         help="Number of values to trust for X-Forwarded-For.")
+        group.add_option("--proxy-x-proto", dest="proxy_x_proto", type="int", my_default=1,
+                         help="Number of values to trust for X-Forwarded-Proto.")
+        group.add_option("--proxy-x-host", dest="proxy_x_host", type="int", my_default=1,
+                         help="Number of values to trust for X-Forwarded-Host.")
```

Comment:
Alternatively, the non-breaking change would be to allow users to opt out of the dangerous behavior with `--proxy-x-host=0`. This is what I am currently doing in my development environment.